### PR TITLE
revised fno

### PR DESCRIPTION
**Pull request by yairchn**
this revised FNO function uses OperatorFlux.jl v0.1.2 and has added assertions for the correct architecture

### CODE REVIEW

File: integration_tests/Project.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ Profile = "9abbd945-dff8-562f-b5e8-e1ebf5ef1b79"
 RRTMGP = "a01a1ee8-cea4-48fc-987c-fc7878d79da1"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 ReportMetrics = "c1654acf-408b-4272-96ce-66c258df8a6c"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```
Comment (charleskawczynski) [inline integration_tests/Project.toml:33]: Let's remove this

Comment (charleskawczynski) [reply integration_tests/Project.toml:33]: And `git checkout main integration_tests/Manifest.toml` to undo those changes.

**Review by charleskawczynski — COMMENTED** [review]
Code changes look fine to me, we should just remove `Revise` and undo the resulting manifest file changes.

**Review by charleskawczynski — APPROVED** [review]
LGTM

**Review by charleskawczynski — APPROVED** [review]
Code changes look fine to me.

**Comment by zongyi-li** [comment]
It looks good to me too.

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
Bors r-

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
Bors r+

**Comment by charleskawczynski** [comment]
Bors r+

**Pull request merged**